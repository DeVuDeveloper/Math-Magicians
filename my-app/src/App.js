import Calculator from './components/Calculator';