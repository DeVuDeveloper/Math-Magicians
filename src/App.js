import react from 'react';
import Calculator from './components/Calculator';

class App extends react.Component {
  render() {
    return <Calculator />;
  }
}

export default App;
