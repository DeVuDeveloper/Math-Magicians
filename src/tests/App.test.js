import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../App';

const Mock = () => (

  <App />

);

describe('testing components', () => {
  it('if render Title', () => {
    render(<App />);
    screen.getByText('Welcome to my page !');
  });
  it('if link calculator works ', () => {
    render(<App />);

    fireEvent.click(screen.getByText('Calculator'));
    screen.getByText("Let's do some math!");
  });
  it('if link Quotes works', () => {
    render(<App />);
    fireEvent.click(screen.getByText('Quotes'));
    screen.getByText('Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston');
  });
  it('if App is displaying right', () => {
    const display = renderer
      .create(<Mock />)
      .toJSON();
    expect(display).toMatchSnapshot();
  });
});
