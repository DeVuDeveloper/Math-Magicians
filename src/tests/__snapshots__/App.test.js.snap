// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing components if App is displaying right 1`] = `
<div>
  <nav
    className="navbar"
  >
    <h1
      className="header"
    >
      Math Magicians
    </h1>
    <ul
      className="link-list"
    >
      <li
        className="link"
      >
        <a
          activeclassname="active"
          className="link"
          exact="true"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "borderBottom": "2px solid #376e6f",
              "paddingBottom": "5px",
            }
          }
        >
          Home
        </a>
      </li>
      <li
        className="link"
      >
        <a
          activeclassname="active"
          className="link"
          exact="true"
          href="/calculator"
          onClick={[Function]}
          style={
            Object {
              "borderBottom": "2px solid #376e6f",
              "paddingBottom": "5px",
            }
          }
        >
          Calculator
        </a>
      </li>
      <li
        className="link"
      >
        <a
          activeclassname="active"
          aria-current="page"
          className="link active"
          exact="true"
          href="/quote"
          onClick={[Function]}
          style={
            Object {
              "borderBottom": "3px solid #fbeec1",
              "paddingBottom": "10px",
            }
          }
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="quote-wrapper"
  >
    <h2
      className="quote"
    >
      Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
    </h2>
  </div>
</div>
`;
